### PR TITLE
Porting Dam to pybind

### PR DESCRIPTION
**Pull request by ipouplana**
Porting Dam Application to pybind11 as requested in #1728.

### CODE REVIEW

File: applications/DamApplication/CMakeLists.txt [hunk 1]
```diff
@@ -2,131 +2,115 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 message("**** configuring KratosDamApplication ****")
 
-include_directories( ${CMAKE_SOURCE_DIR}/kratos )
+################### PYBIND11
+include(pybind11Tools)
 
-#dependecies on other applications
+include_directories( ${CMAKE_SOURCE_DIR}/kratos )
```
Comment (RiccardoRossi) [inline applications/DamApplication/CMakeLists.txt:8]: this should already be included

File: applications/DamApplication/CMakeLists.txt [hunk 2]
```diff
@@ -137,13 +121,16 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
...
-if(${INSTALL_PYTHON_FILES} MATCHES ON)
-  get_filename_component (CURRENT_DIR_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME)
-  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/python_scripts DESTINATION applications/${CURRENT_DIR_NAME}  FILES_MATCHING PATTERN "*.py"  PATTERN ".svn" EXCLUDE)
-endif(${INSTALL_PYTHON_FILES} MATCHES ON)
+###############################################################################
+## installing the resulting libraries
 
-# get_property(inc_dirs DIRECTORY PROPERTY INCLUDE_DIRECTORIES)
-# message("TestApplication subdir inc_dirs = ${inc_dirs}")
+install(TARGETS KratosDamCore DESTINATION libs )
```
Comment (RiccardoRossi) [inline applications/DamApplication/CMakeLists.txt]: you miss the "cotire" take a look to the other apps

**Review by RiccardoRossi — CHANGES_REQUESTED** [review]
Hi @ipouplana take a look to my comments.

at least the one about "cotire" is important

File: applications/DamApplication/CMakeLists.txt (hunk 1 above)
Comment (ipouplana) [reply applications/DamApplication/CMakeLists.txt:8]: What do you mean?

File: applications/DamApplication/CMakeLists.txt (hunk 2 above)
Comment (ipouplana) [reply applications/DamApplication/CMakeLists.txt]: Thanks, I'll add it both for the Poromechanics and the Dam CMakeLists

File: applications/DamApplication/CMakeLists.txt (hunk 1 above)
Comment (roigcarlo) [reply applications/DamApplication/CMakeLists.txt:8]: I don't see anything wrong with that line either :S

Comment (ipouplana) [reply applications/DamApplication/CMakeLists.txt:8]: @RiccardoRossi ping! ;)

Comment (RiccardoRossi) [reply applications/DamApplication/CMakeLists.txt:8]: i am just telling that i think it is included automatically

if i am wrong please approve and go ahead

Comment (ipouplana) [reply applications/DamApplication/CMakeLists.txt:8]: @roigcarlo could you approve this PR? ^_^'

File: applications/DamApplication/CMakeLists.txt [hunk 3]
```diff
@@ -2,131 +2,115 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
...
+        ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_constitutive_laws_to_python.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/dam_python_application.cpp
     )
     
-    ## define library Kratos which defines the basic python interface
-    add_library(KratosDamApplication SHARED ${KRATOS_DAM_APPLICATION_SOURCES})
-    target_link_libraries(KratosDamApplication KratosCore KratosPoromechanicsApplication)
+    ## DamApplication core library (C++ parts)
+    add_library(KratosDamCore SHARED ${KRATOS_DAM_APPLICATION_CORE_SOURCES})
+    target_link_libraries(KratosDamCore PRIVATE KratosCore KratosPoromechanicsCore)
```
Comment (roigcarlo) [inline applications/DamApplication/CMakeLists.txt:96]: I detected that windows requires That one to be

```
target_link_libraries(KratosDamCore PUBLIC KratosCore KratosPoromechanicsCore)
```

But let me speak this with the @KratosMultiphysics/technical-committee first

Comment (ipouplana) [reply applications/DamApplication/CMakeLists.txt:96]: Ok, thanks. If that’s the case I’ll modify it.

Comment (ipouplana) [reply applications/DamApplication/CMakeLists.txt:96]: In the end, should I use PUBLIC or PRIVATE?

**Review by RiccardoRossi — APPROVED** [review]

File: applications/DamApplication/CMakeLists.txt (hunk 3 above)
Comment (roigcarlo) [reply applications/DamApplication/CMakeLists.txt:96]: Public, but don't worry we will change this later on. Right now it works on linux which is enough for us to make the merge.

**Comment by roigcarlo** [comment]
I am gonna cherry pick a change from #1803 so travis pases

File: applications/DamApplication/CMakeLists.txt (hunk 3 above)
Comment (ipouplana) [reply applications/DamApplication/CMakeLists.txt:96]: Ok, thanks @roigcarlo !

**Pull request merged**